### DISCUSSION TITLE
ABRP live data issues

### DISCUSSION THREAD

**Discussion by estragonator** [tillsteinbach/VWsFriend#493] [Q&A] [answered]
Hello everyone,
I have been using ABRP with VWsFriend as a source for live data a few times in the past and it worked generally quite well - however I have tried it again yesterday and I noticed the SoC not updating in the routeplanner app. The routeplanner app still shows the api status as "Connected" and the "latest live data connection" timestamp is updated, but the values of SoC, Odometer stay the same...

I was wondering where the issue is, has anyone experienced simmilar things in the past?

I have updated VWsFriend to 0.23.10, deleted and remade the api connection token and the vehicle in ABRP and was able to reproduce the behavior still today. The recorded data in VWsFriend looks fine, it shows the changes in SoC over time correctly. From the logs I can see that something is sent to ABRP the whole time as well since I get a message like that every two minutes:
`2023-05-21T14:33:21+0000:INFO:abrp_agent:ABRP send telemetry for vehicle xxx for account yyy: Missing telemetry: speed, `

**Comment by estragonator** [comment]
After some more testing I do think there might be an issue with the way battery soc is sent to ABRP during a trip.
I have done testing with a simple rest call from my home automation system to send the soc and the updates in the App worked perfectly.
The main difference (I assume) is the frequency of updates where I sent a hardcoded "every 10 seconds" even the data does not update as fast. Also the timestamp sent was not the utc timestamp recieved from the weconnect api, but just the current utc time.

Would there be a way to enable debug/check what exactly VWsFriend sends to ABRP, to figure out how to improve this behaviour in the future?

**Reply by tillsteinbach** [reply]
Yes, I can include the data in the log output. Will do so in the next version.

**Reply by estragonator** [reply, marked as answer]
I have finally had achance to retest and the functionality is back to working, thanks for your effort! I appreciate having the whole data visible in the log when it is sent to abrp.
